service/key.go

```diff
@@ -53,7 +53,8 @@ func (svc service) CreateKey(_ context.Context, keyCreateReq model.KeyRequest) (
 
 	var err error
 	var createdKey *model.KeyResponse
-	if keyCreateReq.KeyInfo.KeyData == "" && keyCreateReq.KeyInfo.KmipKeyID == "" {
+	if keyCreateReq.KeyInfo.KeyData == "" &&
+		(keyCreateReq.KeyInfo.KmipKeyID == "" || keyCreateReq.KeyInfo.OciSecretId == "") {
```

Comment:
assuming OciSecretId is meant for Register operation, OR should be replaced with AND.

Reply:
Ah yes, good catch.